base/src/main/java/org/geysermc/api/util/ApiVersion.java

```diff
@@ -0,0 +1,91 @@
+package org.geysermc.api.util;
+
+/**
+ * Represents a version of the api.
+ */
+public class ApiVersion {
+    private final int human;
+    private final int major;
+    private final int minor;
+
+    public ApiVersion(int human, int major, int minor) {
+        this.human = human;
+        this.major = major;
+        this.minor = minor;
+    }
+
+    /**
+     * Returns the human version of the api.
+     *
+     * @return the human version
+     */
+    public int human() {
+        return this.human;
+    }
+
+    /**
+     * Returns the major version of the api.
+     *
+     * @return the major version
+     */
+    public int major() {
+        return this.major;
+    }
+
+    /**
+     * Returns the minor version of the api.
+     *
+     * @return the minor version
+     */
+    public int minor() {
+        return this.minor;
+    }
+
+    /**
+     * Checks whether the requested version is compatible with this version.
+     * The parameters represent the desired api version, which is checked against this version.
+     * The human version must match, and the desired major version must be equal or smaller than this major version.
+     * If the major versions match, then the desired minor version must be equal or smaller than this minor version.
+     *
+     * @param human the desired human version
+     * @param major the desired major version
+     * @param minor the desired minor version
+     * @return a {@link Compatibility} indicating whether the versions are compatible
+     */
+    public Compatibility supportsRequestedVersion(int human, int major, int minor) {
```

Comment:
```suggestion
    /**
     * Checks whether an API described by this version has a good likelihood of supporting an API consumer.
     * <p>
     * The compatibility is described as follows:
     * The human version must match.
     * The requested major version must be equal or smaller than this major version.
     * If the major versions are equal, the requested minor version must be equal or smaller than this minor version.
     *
     * @param human the requested human version
     * @param major the requested major version
     * @param minor the requested minor version
     * @return a {@link Compatibility} indicating whether this version may support the requested version
     */
    public Compatibility supportsRequestedVersion(int human, int major, int minor) {
```
reworded the terminology a bit to make the relationship between this and the params clearer.
i also mentioned that there is a "good likelihood" of support, because if they need deprecated methods that have been removed in the current API version, this method might return true, but those methods won't work.